src/main/java/org/truffleruby/core/rope/RopeNodes.java

```diff
@@ -1139,15 +1139,6 @@ public boolean ropesEqual(Rope a, Rope b,
             return a.getEncoding() == b.getEncoding() && a.byteLength() == b.byteLength() && a.hashesMatch(b) && Arrays.equals(aBytes.execute(a), bBytes.execute(b));
         }
 
-        @Specialization(guards = "!isRope(b)")
-        public boolean ropeEqualNonRope(Rope a, Object b) {
```

Comment:
What's wrong here? This was intended to match the following fragment in `Rope#equals`:

```
if (o instanceof Rope) {
  // Actual equality check.
}

return false;
```